### PR TITLE
[postgresql] unique cursor name

### PR DESCRIPTION
**Pull request by MorriganR**
the non-unique cursor name is confusing

before:

after:

### CODE REVIEW

File: src/providers/postgres/qgspostgresconn.h [hunk 1]
```diff
@@ -546,7 +546,7 @@ class QgsPostgresConn : public QObject
     bool mSwapEndian;
     void deduceEndian();
 
-    int mNextCursorId;
+    static int mNextCursorId;
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresconn.h:549]: I think this will need to be QAtomicInt, or something else with thread safety...

Comment (MorriganR) [reply src/providers/postgres/qgspostgresconn.h]: Thanks for the tip.

File: src/providers/postgres/qgspostgresconn.cpp [hunk 2]
```diff
@@ -298,6 +298,8 @@ static void noticeProcessor( void *arg, const char *message )
   QgsMessageLog::logMessage( QObject::tr( "NOTICE: %1" ).arg( msg ), QObject::tr( "PostGIS" ) );
 }
 
+QAtomicInt QgsPostgresConn::mNextCursorId = 0;
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresconn.cpp:301]:
```suggestion
QAtomicInt QgsPostgresConn::sNextCursorId = 0;
```

File: src/providers/postgres/qgspostgresconn.h [hunk 3]
```diff
@@ -546,7 +546,7 @@ class QgsPostgresConn : public QObject
     bool mSwapEndian;
     void deduceEndian();
 
-    int mNextCursorId;
+    static QAtomicInt mNextCursorId;
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresconn.h:549]:
```suggestion
    static QAtomicInt sNextCursorId;
```

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks! This will indeed make things much easier to understand...

File: src/providers/postgres/qgspostgresconn.cpp [hunk 4]
```diff
@@ -1601,7 +1602,6 @@ bool QgsPostgresConn::closeCursor( const QString &cursorName )
 
 QString QgsPostgresConn::uniqueCursorName()
 {
-  QMutexLocker locker( &mLock ); // to protect access to mNextCursorId
   return QStringLiteral( "qgis_%1" ).arg( ++mNextCursorId );
```
Comment (nyalldawson) [inline src/providers/postgres/qgspostgresconn.cpp:1605]:
```suggestion
  return QStringLiteral( "qgis_%1" ).arg( ++sNextCursorId );
```

**Pull request merged**